cmd/nerdctl/ps_linux_test.go

```diff
@@ -0,0 +1,180 @@
+
+	lines := strings.Split(stdout, "\n")
+
+	if len(lines) < 2 {
+		err = fmt.Errorf("stdout should have at least two lines, but it is: %s", stdout)
+		return
+	}
+
+	headerLine, containerLine = lines[0], lines[1]
+
+	// replace "CONTAINER ID" by "CONTAINER_ID" for split line to fileds
+	headerLine = strings.ReplaceAll(headerLine, "CONTAINER ID", "CONTAINER_ID")
+	headerFields = strings.Fields(headerLine)
+	containerFields = strings.Fields(containerLine)
+	return
+}
+
+func getValueByReference(header, container, title, nextTitle string) string {
+	var startIdx, endIdx int
+	if title == "" {
+		startIdx = 0
+	} else {
+		startIdx = strings.Index(header, title)
+	}
+	if nextTitle == "" {
+		endIdx = len(container)
+	} else {
+		endIdx = strings.Index(header, nextTitle)
+	}
+
+	if startIdx < 0 || endIdx < 0 || endIdx > len(container) {
+		return ""
+	}
+
+	return strings.TrimSpace(container[startIdx:endIdx])
+}
+
+func TestContainerList(t *testing.T) {
+	base := prepareTest(t)
+	defer base.Cmd("rm", "-f", testContainerListName).AssertOK()
+
+	// hope there are no tests running parallel
+	base.Cmd("ps", "-n", "1", "-s").AssertOutWithFunc(func(stdout string) error {
+		// An example of nerdctl/docker ps -n 1 -s
+		// CONTAINER ID    IMAGE                               COMMAND    CREATED           STATUS    PORTS    NAMES            SIZE
+		// be8d386c991e    docker.io/library/busybox:latest    "top"      1 second ago    Up                 c1       16.0 KiB (virtual 1.3 MiB)
+		headerLine, containerLine, headerFields, containerFields, err := parseStdout(stdout)
+		if err != nil {
+			return err
+		}
+
+		expectedHeaders := []string{"CONTAINER_ID", "IMAGE", "COMMAND", "CREATED", "STATUS", "PORTS", "NAMES", "SIZE"}
+		if !reflect.DeepEqual(expectedHeaders, headerFields) {
+			return fmt.Errorf("expect header %+v, but got %+v", expectedHeaders, headerFields)
+		}
+
+		if testutil.CommonImage != containerFields[1] {
+			return fmt.Errorf("expect container image %s, but got %s", testutil.CommonImage, containerFields[1])
+		}
+
+		containerName := getValueByReference(headerLine, containerLine, "NAMES", "SIZE")
+		if testutil.CommonImage != containerFields[1] {
+			return fmt.Errorf("expect container name %s, but got %s", testContainerListName, containerName)
+		}
+
+		actualSize := getValueByReference(headerLine, containerLine, "SIZE", "")
+
+		// there is some difference between nerdctl and docker in calculating the size of the container
+		expectedSize := "26.2MB (virtual "
+		if base.Target != testutil.Docker {
+			expectedSize = "25.0 MiB (virtual "
+		}
+
+		if !strings.Contains(actualSize, expectedSize) {
+			return fmt.Errorf("expect container size %s, but got %s", expectedSize, containerName)
+		}
+
+		return nil
+	})
+
+}
+
+func TestContainerListWideMode(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := prepareTest(t)
+	defer base.Cmd("rm", "-f", testContainerListName).AssertOK()
```

Comment:
This can be moved to prepareTest and under prepareTest; you can call
`t.Cleanup = func() { base.cmd(...) }`

Out of curiosity: is the AssertOk useful? I am not sure

Reply:
If AssertXXX is not called, the cmd will not be executed, of course, there is also a `Run()` that can be used to run the command.